### ISSUE TITLE
[Bug] trashed users aren't showing up in select2_from_ajax field

### ISSUE THREAD

**Issue by prescriptionlifeline** [Laravel-Backpack/community-forum#1252]
# Bug report

### What I did

In my User CRUD I have this:

```php
        CRUD::field([
            'name' => 'spouse',
            'label' => 'Spouse',
            'type' => 'select2_from_ajax',
            'entity' => 'spouse',
            'attribute' => 'lastNameFirstName',
            'data_source' => backpack_url('user/fetch/spouse'),
            'method' => 'POST',
        ]);
```

Here' my FetchOperation:

```
    protected function fetchSpouse()
    {
        return $this->fetch([
            'model' => User::class, // required
            'searchable_attributes' => ['lastNameFirstName'],
            'paginate' => 10, // items to show per page
            'searchOperator' => 'LIKE',
            'query' => function($model) {
                //return $model->withTrashed();
                return $model->withTrashed()->where(function($query) {
                    $query->whereNull('deleted_at')
                          ->orWhereNotNull('deleted_at');
                });
            } // to filter the results that are returned
        ]);
    }
```
And here's my spouse method in my User model:

```php
    public function spouse()
    {
        return $this->belongsTo(User::class, 'spouseID')->withTrashed();
    }
```

### What I expected to happen

With all of this I can _add_ trashed spouses to non trashed users without issue. My expectation is that the newly added members would then show up when I refresh the page.

If I do `User::findOrFail(...)->spouse` in `php artisan tinker` that returns the spouse even if the spouse isn't trashed. I just can't get the `select2_from_ajax` to behave the same way as `php artisan tinker`.

### What happened

After adding a trashed spouse to a non-trashed user and then refreshing the trashed spouse does _not_ show up.

### What I've already tried to fix it

I posted https://github.com/Laravel-Backpack/community-forum/discussions/1139 but I think at this point this prob needs to be seen as a bug in Laravel Backpack.

### Is it a bug in the latest version of Backpack?

> After I run ```composer update backpack/crud``` the bug... is it still there?

Yes

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:

```
### PHP VERSION:
8.3.3

### PHP EXTENSIONS:
Core, bcmath, calendar, ctype, date, filter, hash, iconv, json, SPL, pcre, random, readline, Reflection, session, standard, mysqlnd, tokenizer, zlib, libxml, dom, PDO, openssl, SimpleXML, xml, xmlreader, xmlwriter, curl, gmp, pdo_mysql, mysqli, fileinfo, sqlite3, pdo_sqlite, mbstring, Phar, sodium

### LARAVEL VERSION:
10.48.20.0

### BACKPACK PACKAGE VERSIONS:
backpack/basset: 1.3.5
backpack/crud: 6.7.43
backpack/filemanager: 3.0.8
backpack/generators: v4.0.5
backpack/pro: 2.2.14
backpack/theme-tabler: 1.2.11
```

**Comment by jcastroa87** [comment]
Hello @prescriptionlifeline if you see the data save on table, there is the spouse value?

Cheers.

**Comment by jcastroa87** [comment]
Hello @prescriptionlifeline, I found the issue here.

To solve, you can override the select2_from_ajax field, following [these steps](https://backpackforlaravel.com/docs/6.x/crud-fields#overwriting-default-field-types-1)

Then replace this line:
```
$item = $connected_entity->find($old_value);
```

With:
```
$item = $connected_entity->withTrashed()->find($old_value);
```

That's should work.

Cheers.

**Comment by prescriptionlifeline** [comment]
I'll try that out after I get to work - thanks!

**Comment by prescriptionlifeline** [comment]
I wasn't able to get this working.

The only file that has `$item = $connected_entity->find($old_value);` in it is vendor/pro/resources/views/fields2/select2_from_ajax.blade.php. As a sanity check I first tried to modify that file as you suggested. After making said changes I did `php artisan cache:clear` and then reloaded the page and the trashed user was still not showing up.

I then renamed that file to zzz.select2_from_ajax.blade.php and did `php artisan cache:clear` and the page loads without issue, like it's not even looking at that file when it's building the page.

I then similarly renamed _every_ select2* file and did `php artisan cache:clear` and, once again, the page loaded just fine...

**Comment by jcastroa87** [comment]
Hi @prescriptionlifeline

There are many errors here.

First, there is no "fields2" folder in vendor, it is just "fields".
Then you didn't follow instructions on our docs, you should not modify file in vendor, need to publish them:
```
php artisan backpack:field --from=select2_from_ajax
```

I try on my side and is working, here are the examples:

I create Tag (Spouses) and Category (User), both use Soft Deletes.

So in "Category" Controller I add this:
```
...
use \Backpack\CRUD\app\Http\Controllers\Operations\FetchOperation;
...

protected function setupCreateOperation()
    {
        ...
        CRUD::addField([
            'name' => 'tag_id',
            'label' => 'Tag',
            'type' => 'select2_from_ajax',
            'entity' => 'tag',
            'attribute' => 'name',
            'data_source' => backpack_url('category/fetch/tag'),
            'placeholder' => 'Select a tag',
            'minimum_input_length' => 0,
            'method' => 'POST',
        ]);
    }

...

protected function fetchTag()
    {
        return $this->fetch([
            'model' => \App\Models\Tag::class,
            'searchable_attributes' => ['name'],
            'paginate' => 10,
            'searchOperator' => 'LIKE',
            'query' => function($model) {
                return $model->withTrashed();
            }
        ]);
    }
```

In "Category" Model:
```
...
public function tag()
    {
        return $this->belongsTo(Tag::class)->withTrashed();
    }
```

I publish "select2_from_ajax"
```
php artisan backpack:field --from=select2_from_ajax
```

And replace
```
$item = $connected_entity->find($old_value);
```

With
```
$item = $connected_entity->withTrashed()->find($old_value);
```

Now here are the screenshots:

Item "test2" from Tag is deleted
![image](https://github.com/user-attachments/assets/2dc96ea9-a9d4-46b3-bcda-84e61a40e88d)

In Category i still can see "test2"
![image](https://github.com/user-attachments/assets/c923fd19-3a44-48d8-902b-3f1ca3202be8)

I save the Category
![image](https://github.com/user-attachments/assets/e818d7d1-fb63-4135-9381-29322a3b72fa)

I see the Category and deleted Tag on list
![image](https://github.com/user-attachments/assets/23962443-4518-4ec5-9e8a-965c47ede035)

When I edit, there still the deleted Tag
![image](https://github.com/user-attachments/assets/24e0ad9d-34a8-48fe-a78d-9ef502c80eba)

Please be careful and follow all the steps in our docs, you should not just replace the files in vendor and if you want to customize the Trashed Operation need to follow the steps in these docs.

Cheers.

**Comment by prescriptionlifeline** [comment]
> First, there is no "fields2" folder in vendor, it is just "fields".

I copied the fields folder to fields2 before I made my changes. I guess I copy / pasted the wrong path when I made my post. I guess everything I've ever done is now forever tainted by that and I should just throw in the towel and start flipping burgers at McDonalds 🙄

> Then you didn't follow instructions on our docs, you should not modify file in vendor, need to publish them****

As I said in the post you're replying to, I changed that file "**_As a sanity check_**".  Allow me to give another example of the concept of a sanity check.

One time, many years ago, I thought I was on the prod server. I was trying to make a change directly to that server and nothing I did was showing up. I then decided, as a sanity test, I just wanted to make _something_ show up. So then I started renaming directories. I renamed the parent directory and nothing. I renamed the `includes/` directory and the website still worked. I then renamed the DocumentRoot directory that the httpd.conf file was pointing to and the website still worked. Turns out I was connected to a dev server instead of the prod server that I thought I was connected.

And you're right - I shouldn't have to modify the vendor directory. That wasn't the point of my tests. [As observed in another post](https://github.com/Laravel-Backpack/community-forum/discussions/1018#discussioncomment-9744798), not everything can be published. Sometimes you just need to straight up copy files. I was modifying the file in the `vendor/` directory , **as a sanity check**, because I wanted to see if that worked **before** trying and possibly failing to publish (https://backpackforlaravel.com/docs/6.x/crud-fields#overwriting-default-field-types-1 mentions being able to publish to 
`/resources/views/vendor/backpack/crud/fields` but not necessarily to `/resources/views/vendor/backpack/pro/fields`). Like if publishing didn't work (which, as has already been established, isn't always an option) I could then manually copy it but then what if I got the directory I was manually copying it to wrong? Like if I'm trying to debug why shit isn't working I'd just assume reduce the number of sources for error rather then maximize them.

Anyway, I'll take a look at the rest of your post later today. The thoroughness of it does look encouraging - the less I have to fill in the gaps the less likely I am to encounter errors.

**Comment by prescriptionlifeline** [comment]
I figured out the issue.

So in the `setupCreateOperation()` method of my UserCrudController I had this:

```php
        CRUD::field([
            'name' => 'spouse',
            'label' => 'Spouse',
            'type' => 'select2_from_ajax',
            'entity' => 'spouse',
            'attribute' => 'name',
            'data_source' => backpack_url('user/fetch/spouse'),
            'method' => 'POST',
            'hint' => 'Search by last name <i>or</i> first name. Adding a spouse to this account will also result in this person being added as a spouse to the spouse\'s account.',
        ]);
```

In the `setupUpdateOperation()` method I was calling `$this->setupCreateOperation()` and was then doing this:

```
        CRUD::field([
            'name' => 'spouse',
            'data_source' => backpack_url('user/fetch/spouse?exclude=' . $this->crud->getCurrentEntry()->id),
            'hint' => 'Search by last name <i>or</i> first name. Adding a spouse to this account will also result in this person being added as a spouse to the spouse\'s account.',
        ]);
```
If it weren't for the `data_source` bit, which was changed per https://github.com/Laravel-Backpack/community-forum/discussions/1131, it wouldn't be necessary to do any additional configuring of the spouse field in `setupUpdateOperation()`.

Anyway, it turns out that adding `'type' => 'select2_from_ajax',` to the spouse field in `setupUpdateOperation()` does the trick.

Like I had been assuming that array keys left undefined would be inherited from the `CRUD::field()` calls in `setupCreateOperation()` but I guess I was wrong. I mean, sometimes, `setupUpdateOperation()` seemed to inherit from the `CRUD::field()` calls in `setupCreateOperation()` but sometimes it didn't. For example, I also have this in `setupCreateOperation()`:

```php
        CRUD::field([
            'name' => 'password',
            'type' => 'password',
            'attributes' => ['disabled' => 'disabled'],
            'hint' => 'This field is disabled because <a tabindex="-1" href="/admin/setting/0/edit#new-user-email" target="_blank">an email with a randomly generated password</a> (eight character alpha-numeric password) will be sent out when the Save button is clicked (unless you uncheck the checkbox at the bottom of this page)'
        ]);
```
...and then I have this in `setupUpdateOperation()`:

```php
        CRUD::field([
            'name' => 'password',
            'attributes' => ['id' => 'password'],
            'hint' => "<span id='passwordHint'>$hint</span>",
        ]);
```
Like even though I'm not telling Laravel Backpack that the type is password in the `setupUpdateOperation` call it just knows. I assumed that that was due to it falling back on previously submitted option but I guess that's not the case. Like I just tried changing type to text for the password field in `setupCreateOperation` and that change did _not_ get reflected in `setupUpdateOperation`. So I guess when the type parameter isn't present it infers it and it may not always infer it correctly.

TLDR I was making some unfounded assumptions about how the whole thing worked and your initial reply did contain the answer.

I still stand by the sanity checks I performed but that's neither here nor there.

Thank you for your time.

**Comment by pxpm** [comment]
This is worth investigating. 
In my opinion you are right, attributes should be inherited from the first field definition. If you have the field in `setupCreateOperation` and you do `$this->setupCreateOperation()` in your update operation is totally expected that the exact same field definitions are set, and you should be able to change only the things you need. 

I've re-opened this discussion so that it can be investigated a bit more 👍 

Thanks for your time and for providing such a good explanation of the issue 🙏 

Cheers

**Comment by pxpm** [comment]
I've just merged a fix for this in `backpack/crud`. I will be tagging a version with the fix later today or tomorrow 👍
